### ISSUE TITLE
Fix intermittent spec error

### ISSUE THREAD

**Issue by oliverbarnes** [BlueRidgeLabs/kimball#44]
[spec/controllers/v2/sms_reservations_controller_spec.rb:24](https://github.com/BlueRidgeLabs/kimball/blob/35-scheduling-call-over-sms/spec/controllers/v2/sms_reservations_controller_spec.rb#L24) breaks from time to time. Might be because of conflicts between time slot factory-generated times and time overlap detection

**Comment by cromulus** [comment]
I wrapped it with a retry and it seems to work now. I think there may be some race conditions in there somewhere.

**Comment by cromulus** [comment]
Nope, this is definitely a bug.

From what I can tell, this is what happens: The event isn't saved because the time slots aren't saved, however, event_invitation continues along, happy with it's invalid event. It only occurs occasionally.

I think perhaps the right move is to delete events all together and associate time_slots with event_invitations. potentially renaming event_invitations to events. or interviews. or some such.

as it stands, event has a one-to-one relationship with event_invitation, stores a user_id and a description, and is where time_slots associate. Seems like it's not really needed and adds complexity.

**Comment by cromulus** [comment]
if it's any use, this is how I tracked it down.
```
ei,p,ts,r = true,true,true,true

def didnt_raised_error?(arr)
  return true if arr[0] == true
  begin
    arr.each(&:save).each(&:reload)
    return true
  rescue Exception
    return false
  end
end

while didnt_raised_error?([ei,p,ts,r])
  ei = FactoryGirl.create(:event_invitation)
  p = ei.invitees.sample
  ts = ei.event.time_slots.sample
  r = V2::Reservation.create(person: p, time_slot: ts)
end
```

**Comment by cromulus** [comment]
fixed in #64
